Doc/library/functions.rst

```diff
@@ -340,8 +340,12 @@ are always available.  They are listed here in alphabetical order.
    It is needed to unambiguous :ref:`filter <warning-filter>` syntax warnings
    by module name.
 
-   This function raises :exc:`SyntaxError` if the compiled source is invalid,
-   and :exc:`ValueError` if the source contains null bytes.
+   This function raises the following exceptions:
+
+   - :exc:`SyntaxError` if the compiled source is invalid
+   - :exc:`ValueError` if the source contains null bytes
+   - :exc:`UnicodeDecodeError` if the source code contains docstrings that
```

Comment:
I wouldn't try to keep this list precise and complete. The exceptions change over time, and it's not like people are going to write except blocks for these specific exceptions.  Details are in the versionchanged blocks below. Here can we say that SyntaxError happens if the source is invalid, and other exceptions if the text can't be read in various ways?